components/custom_board/tas5805m/tas5805m.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * ESPRESSIF MIT License
+ *
+ * Copyright (c) 2020 <ESPRESSIF SYSTEMS (SHANGHAI) CO., LTD>
+ *
+ * Permission is hereby granted for use on all ESPRESSIF SYSTEMS products, in
+ * which case, it is free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the "Software"), to deal in the
+ * Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ *
+ */
+
+#include "tas5805m.h"
+#include "board.h"
+#include "esp_log.h"
+#include "i2c_bus.h"
+#include "tas5805m_reg_cfg.h"
+
+static const char *TAG = "TAS5805M";
+
+#define TAS5805M_ADDR 0x5c
+// #define TAS5805M_RST_GPIO get_pa_enable_gpio ()
+#define TAS5805M_VOLUME_MAX 254
+#define TAS5805M_MUTE 255
+#define TAS5805M_VOLUME_MIN 0
+
+uint8_t* VolumeLevel = 0;
```

Comment:
why do you use a pointer here? shouldn't it be `static uint8_t VolumeLevel = 0` ?

Reply:
please implement this change